README.md

```diff
@@ -18,8 +18,84 @@ npm install geckoboard
 
 ## Use
 
-The latest documentation and user guide can be found on the Geckoboard developer docs (https://developer.geckoboard.com/)
+The latest documentation and user guide can be found on the Geckoboard developer docs (https://developer.geckoboard.com/) 
 
+
+### API requests
+
+#### Find or create a new dataset
+
+```
+import Geckoboard from 'geckoboard';
+
+const API_KEY = 'YOUR_API_KEY';
+
+const gb = new Geckoboard(API_KEY);
+
+const dataset = gb.defineDataset({
+    id: 'my.dataset',
+    fields: {
+        count: {
+            type: 'number',
+            name: 'Count'
+        },
+        day: {
+            type: 'date',
+            name: 'Day'
+        },
+    },
+    uniqueBy: ['day']
+})
+
+const schema = await dataset.create();
+console.log(schema)
+```
+
+#### Update a dataset
+```
+
+await dataset.post([{ count: 1, day: '2023-10-10' }])
+
+// provide an optional 'delete_by' value, to indicate 
+// which fields should be used to indicate which records
+// should be deleted
+await dataset.append([{ count: 3, day: '2023-10-11' }], 'day')
+```
+
+#### Replace all data in a dataset
+```
+// all current data will be replaced
+await dataset.replace([{ count: 2, day: '2023-10-10' }])
+
+```
+#### Delete a dataset
+```
+
+// remove the dataset completely
+await dataset.delete()
+
+```
+
+### Ping to test connection
+
+```
+import Geckoboard from 'geckoboard';
+
+const API_KEY = 'YOUR_API_KEY';
+
+const gb = new Geckoboard(API_KEY);
+
+const run = async (): Promise<void> => {
```

Comment:
what's the purpose of this extra `run` function?

Reply:
It's not necessary really, it's just you can't use `await` without it being in an async function, so you couldn't just copy and paste this into a file without the wrapping `run` async function.

Reply:
That's a good point, do we need to enforce a minimum Node version in package.json based on whatever features of the language/stdlib/runtime we're using?